session_test.go

```diff
@@ -785,30 +787,56 @@ func TestManyStreams_PingPong(t *testing.T) {
 	wg.Wait()
 }
 
-func TestHalfClose(t *testing.T) {
+func TestCloseRead(t *testing.T) {
 	client, server := testClientServer()
 	defer client.Close()
 	defer server.Close()
 
-	stream, err := client.Open()
+	stream, err := client.OpenStream()
 	if err != nil {
 		t.Fatalf("err: %v", err)
 	}
 	if _, err = stream.Write([]byte("a")); err != nil {
 		t.Fatalf("err: %v", err)
 	}
 
-	stream2, err := server.Accept()
+	stream2, err := server.AcceptStream()
 	if err != nil {
 		t.Fatalf("err: %v", err)
 	}
-	stream2.Close() // Half close
+	stream2.CloseRead()
 
 	buf := make([]byte, 4)
 	n, err := stream2.Read(buf)
+	if n != 0 || err == nil {
+		t.Fatalf("read after close: %d %s", n, err)
+	}
+}
+
+func TestHalfClose(t *testing.T) {
```

Comment:
would be good to also have a code path to test the HalfReset closing path

Reply:
Do you mean `CloseRead`?

Reply:
(there's no half reset)